lib/system/mmdisp.nim

```diff
@@ -214,175 +210,125 @@ elif defined(gogc):
-    result = alloc(size)
-    zeroMem(result, size)
+    result = goMalloc(size.uint)
 
   proc realloc(p: pointer, newsize: Natural): pointer =
-    result = c_realloc(p, newsize)
-    if result == nil: raiseOutOfMem()
+    raise newException(Exception, "not implemented")
 
-  proc dealloc(p: pointer) = c_free(p)
+  proc dealloc(p: pointer) =
+    discard
 
   proc allocShared(size: Natural): pointer =
-    result = c_malloc(size)
-    if result == nil: raiseOutOfMem()
+    result = alloc(size)
 
   proc allocShared0(size: Natural): pointer =
-    result = alloc(size)
-    zeroMem(result, size)
+    result = alloc0(size)
 
   proc reallocShared(p: pointer, newsize: Natural): pointer =
-    result = c_realloc(p, newsize)
-    if result == nil: raiseOutOfMem()
+    result = realloc(p, newsize)
 
-  proc deallocShared(p: pointer) = c_free(p)
+  proc deallocShared(p: pointer) = dealloc(p)
 
   when hasThreadSupport:
     proc getFreeSharedMem(): int = discard
     proc getTotalSharedMem(): int = discard
     proc getOccupiedSharedMem(): int = discard
 
-  const goFlagNoZero: uint32 = 1 shl 3
-  proc goRuntimeMallocGC(size: uint, typ: uint, flag: uint32): pointer {.
-    importc: "runtime_mallocgc", dynlib: goLib.}
-
-  proc goSetFinalizer(obj: pointer, f: pointer) {.
-    importc: "set_finalizer", codegenDecl: "$1 $2$3 __asm__ (\"main.Set_finalizer\");\n$1 $2$3",
-    dynlib: goLib.}
-
   proc newObj(typ: PNimType, size: int): pointer {.compilerproc.} =
-    result = goRuntimeMallocGC(roundup(size, sizeof(pointer)).uint, 0.uint, 0.uint32)
+    writebarrierptr(addr(result), goMalloc(size.uint))
     if typ.finalizer != nil:
       goSetFinalizer(result, typ.finalizer)
 
+  proc newObjRC1(typ: PNimType, size: int): pointer {.compilerRtl.} =
+    writebarrierptr(addr(result), newObj(typ, size))
+
   proc newObjNoInit(typ: PNimType, size: int): pointer =
-    result = goRuntimeMallocGC(roundup(size, sizeof(pointer)).uint, 0.uint, goFlagNoZero)
-    if typ.finalizer != nil:
-      goSetFinalizer(result, typ.finalizer)
+    writebarrierptr(addr(result), newObj(typ, size))
 
   proc newSeq(typ: PNimType, len: int): pointer {.compilerproc.} =
-    result = newObj(typ, len * typ.base.size + GenericSeqSize)
+    writebarrierptr(addr(result), newObj(typ, len * typ.base.size + GenericSeqSize))
     cast[PGenericSeq](result).len = len
     cast[PGenericSeq](result).reserved = len
     cast[PGenericSeq](result).elemSize = typ.base.size
 
+  proc newSeqRC1(typ: PNimType, len: int): pointer {.compilerRtl.} =
+    writebarrierptr(addr(result), newSeq(typ, len))
+
   proc nimNewSeqOfCap(typ: PNimType, cap: int): pointer {.compilerproc.} =
     result = newObj(typ, cap * typ.base.size + GenericSeqSize)
     cast[PGenericSeq](result).len = 0
     cast[PGenericSeq](result).reserved = cap
     cast[PGenericSeq](result).elemSize = typ.base.size
 
+  proc typedmemmove(dest: pointer, src: pointer, size: uint) {.importc: "typedmemmove", dynlib: goLib.}
```

Comment:
``typedMemMove``

Reply:
All problems fixed by amending this commit.